### PR TITLE
Bug #75472, Text Layout Designer, unable to delete entire row

### PR DESCRIPTION
**Pull request by milotalon**
Root Cause:
The Text Layout Designer only provided per-item delete controls (the "X" on each Field/Text/Spacer chip). A row was only removed implicitly, when its last item was deleted. There was no row-level delete action, so the only way to get rid of a row was to manually remove every chip in it one at a time.

Fix:
Added a row-level "X" (Remove Row) button to each row in the Text Layout Designer. Clicking it deletes the entire row and all of its items in one action, and the rows below shift up. Any field bindings used only by that row are dropped from the chart's text fields (a binding that is still referenced by another row is kept), with the remaining field indices compacted to stay in sync. The change is frontend-only (text-layout-designer.component .ts/.html/.scss); the removal is applied to the working copy and only persisted on OK, so Cancel still discards it. Covered by new unit tests.

### CODE REVIEW

File: web/projects/portal/src/app/binding/editor/chart/aesthetic/text-layout-designer.component.ts [hunk 1]
```diff
@@ -398,6 +398,52 @@ export class TextLayoutDesignerComponent implements OnInit, AfterViewInit, OnDes
...
+      if(!row) {
+         return;
+      }
+
+      // Remove the row first; remaining rows shift up via splice.
+      this.workingRows.splice(rowIndex, 1);
+
+      // Unique FIELD indices that were in the removed row, now orphaned only if no surviving row
+      // still references them. Sort descending so removals don't invalidate pending indices.
+      const orphaned = Array.from(new Set(
```
[the remark this answers is not in the document]
Comment (milotalon) [reply web/projects/portal/src/app/binding/editor/chart/aesthetic/text-layout-designer.component.ts:431]: Done — extracted the shared logic into a private `removeOrphanedField(fi)` helper; both `removeFieldItem` and `removeRow` now call it, so the orphan/compaction contract lives in one place. (4a3b1747d)

File: web/projects/portal/src/app/binding/editor/chart/aesthetic/text-layout-designer.component.ts [hunk 2]
```diff
@@ -398,6 +398,52 @@ export class TextLayoutDesignerComponent implements OnInit, AfterViewInit, OnDes
...
+         row.items
+            .filter(it => it.type === this.FIELD && it.fieldIndex != null)
```
[the remark this answers is not in the document]
Comment (milotalon) [reply web/projects/portal/src/app/binding/editor/chart/aesthetic/text-layout-designer.component.ts:433]: Added the defensive `(row.items ?? [])` null-coalesce. (4a3b1747d)

File: web/projects/portal/src/app/binding/editor/chart/aesthetic/text-layout-designer.component.spec.ts [hunk 3]
```diff
@@ -122,4 +122,42 @@ describe("TextLayoutDesignerComponent Unit Test", () => {
...
+
+      component.removeRow(0);
+
+      // field 0 still used by the surviving row — keep it, no emit
+      expect(removed).not.toHaveBeenCalled();
+      expect(component.textFields.length).toBe(1);
+      expect(component.workingRows.length).toBe(1);
+      expect(component.workingRows[0].items[0].fieldIndex).toBe(0);
+   });
 });
```
[the remark this answers is not in the document]
Comment (milotalon) [reply web/projects/portal/src/app/binding/editor/chart/aesthetic/text-layout-designer.component.spec.ts:194]: Added both edge-case tests (row with only STATIC/SPACING items; removing the last remaining row). All 9 specs pass. (4a3b1747d)

File: web/projects/portal/src/app/binding/editor/chart/aesthetic/text-layout-designer.component.html [hunk 4]
```diff
@@ -214,6 +214,11 @@
         </div>
 
       </div>
+
+      <!-- Row-level remove: deletes the entire row and its items (bindings dropped, rows shift up) -->
```
[the remark this answers is not in the document]
Comment (milotalon) [reply web/projects/portal/src/app/binding/editor/chart/aesthetic/text-layout-designer.component.html]: Removed the comment. (4a3b1747d)

**Review by jshobe-inetsoft — APPROVED** [review]
Approved: Claude automated review is complete with no blocking issues, all checks pass, and the branch is mergeable.
